src/deepsparse/evaluation/registry.py

```diff
@@ -17,23 +17,23 @@
 import logging
 from typing import Any, Callable, List, Optional, Union
 
-from sparsezoo.utils.registry import RegistryMixin
+from sparsezoo.evaluation import EvaluationRegistry
 
 
-__all__ = ["EvaluationRegistry"]
+__all__ = ["DeepSparseEvaluationRegistry"]
 
 _LOGGER = logging.getLogger(__name__)
 
 
-class EvaluationRegistry(RegistryMixin):
+class DeepSparseEvaluationRegistry(EvaluationRegistry):
```

Comment:
What in this class is deepsparse specific? Could we move this to the SparseZoo base class?

Reply:
This class has a resolve method, that imports certain modules from deepsparse; I would like to keep importing deepsparse modules in deepsparse and not in sparsezoo unless there's a strong reasoning behind that; the idea being sparsezoo would just define the contract and contain common utilities whereas deepsparse and sparseml would add things relevant to just them.

In this case the partially_check_dependency method is coupled with the integrations. As a future step, we might add a small config file, mapping integration name --> their corresponding files, that could potentially remove the coupling and we can unify more code then?

Reply:
I think @rahul-tuli's idea is good. So we have a "chain of inheritance":
1. `sparsezoo.RegistryMixin`
2. `sparsezoo.EvaluationRegistry`
3. `deepsparse.DeepSparseEvaluationRegistry` / `sparseml.SparseMLEvaluationRegistry`
right?

Reply:
Yes!